lab-kyle/lib/note-crud.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const Note = require('../model/note');
+const AppErr = require('../lib/app-error');
+const debug = require('debug')('awesomeNote:note-crud');
+const userCrud = require('./user-crud');
+
+exports.createNote = function(reqBody){
+  debug('note-crud createNote');
+  return new Promise((resolve, reject) => {
+    if (! reqBody.content)
+      return reject(AppErr.error400('note require\'s content'));
+    if(! reqBody.name)
+      return reject(AppErr.error400('note require\'s name'));
+
+    reqBody.timestamp = new Date();
+    const note = new Note(reqBody);
+    note.save()
+    .then(resolve)
+    .catch(reject);
+  });
+};
+
+exports.fetchNoteByUserId = function(userId){
+  debug('note-crud fetchNoteByUserId');
+  return new Promise((resolve, reject) => {
+    userCrud.fetchUser({_id: userId})
+    .then( note => Note.find({userId: note._id}))
+    .then( notes => resolve(notes))
+    .catch( err => reject(err));
+  });
+};
+
+exports.updateNote = function(id, data){
+  return new Promise((resolve, reject) => {
+    if(!id){
+      var err = AppErr.error400('bad request');
+      return reject(err);
+    }
+    if(!data){
+      err = AppErr.error400('bad request');
+      return reject(err);
+    }
+    if(!data.content){
+      err = AppErr.error400('bad request');
+      return reject(err);
+    }
+
+    Note.findOne({_id: id})
```

Comment:
Here again you are doing an unnecessary lookup. You are looking up a note, then looking up the note again to update it, then looking it up again to get the updated version of it. This could be done like this:

Note.findByIdAndUpdate(id, {$set: data}, {new: true})
.then(resolve)
.catch(err => reject(AppErr.error404(err.message)));